wd_capabilities.au3

```diff
@@ -179,75 +173,72 @@ Func _WD_CapabilitiesAdd($key, $value1 = Default, $value2 = Default)
 	Local Const $s_Parameters_Info = '     $key = ' & $key & '     $value1 = ' & $value1 & '     $value2 = ' & $value2
 
 	If StringRegExp($key, $_WD_KEYS__MATCHTYPES, $STR_REGEXPMATCH) Then ; check if alwaysMatch|firstMatch
-		Local $iResult = __WD_CapabilitiesInitialize($key, $value1)
-		__WD_ConsoleWrite($sFuncName & ": TESTING #" & @ScriptLineNumber & $s_Parameters_Info & "  :: DEBUG", $_WD_DEBUG_Full)
-		If Not @error Then $_WD_CAPS__CURRENTIDX = $iResult
-		Return SetError(@error, @extended, $_WD_CAPS__CURRENTIDX)
+		__WD_CapabilitiesInitialize($key, $value1)
+		__WD_ConsoleWrite($sFuncName & ': #' & @ScriptLineNumber & $s_Parameters_Info, $_WD_DEBUG_Full)
+		Return SetError(@error, @extended)
 	EndIf
-	If $_WD_CAPS__CURRENTIDX = -1 Then _
-			Return SetError(__WD_Error($sFuncName, $_WD_ERROR_GeneralError, "Must be properly initialized"))
 
-	Local $s_SpecificOptions_KeyName = $_WD_CAPS__API[$_WD_CAPS__CURRENTIDX][$_WD_CAPS__SPECIFICVENDOR__ObjectName]
+	Local $bAlwaysMatch_exist = Json_ObjExists($_WD_CAPS__OBJECT, 'capabilities.alwaysMatch')
+	SetError(0) ; for any case because Json_ObjExists() can set @error
+	__WD_ConsoleWrite($sFuncName & ': #' & @ScriptLineNumber & '   $bAlwaysMatch_exist = ' & $bAlwaysMatch_exist, $_WD_DEBUG_Full)
+	If $bAlwaysMatch_exist = 0 Then _
+			Return SetError(__WD_Error($sFuncName, $_WD_ERROR_GeneralError, 'Must be properly initialized'))
```

Comment:
This is a change in behavior that we should discuss. AFAICS, "alwaysMatch" is optional in the specs so why should it be required here?

Reply:
https://www.w3.org/TR/webdriver/#processing-capabilities

`Let required capabilities be the result of getting the property "alwaysMatch" from capabilities request.`

What this sentence mean ?
Especially `Let required capabilities be ` ?

Is that mean be:

`Let required (rquested by user) capabilities be the result of getting the property "alwaysMatch" from capabilities request.`

?

Reply:
> "alwaysMatch" is optional in the specs

Please quote a selected fragment from the specification.

Reply:
> What (does) this sentence mean ?

To me, it simply means that "required capabilities" is defined as the alwaysMatch property from the Capabilities string.

> Is that mean:

Yes, I believe so

> Please quote a selected fragment from the specification.

Read the next line down in the specs (7.2.2.1) 😄

Reply:
> Read the next line down in the specs (7.2.2.1)

It solve my issue with 
```autoit
Json_ObjExists($_WD_CAPS__OBJECT, 'capabilities.alwaysMatch')
```
https://www.autoitscript.com/forum/topic/148114-a-non-strict-json-udf-jsmn/?do=findComment&comment=1499961

Reply:
btw>
I was trying test using wd_demo.au3 with such capabilities
```autoit
	_WD_CapabilitiesStartup()
	Local $sCapabilities = _WD_CapabilitiesGet()
	Return $sCapabilities
```
Chrome, FireFox, Edge works well.

Opera is throwing errors:

```
> _WD_CreateSession
__WD_Post: URL=HTTP://127.0.0.1:9515/session; $sData={
    "capabilities":{
        "alwaysMatch":{}
    }
}
__WD_Post ==> Success (0) HTTP status = 200 : ResponseText={"sessionId":"7834b5cf21aa6af0398dcabe34022b1e","status":33,"value":{"message":"session not created:...
_WD_CreateSession: {"sessionId":"7834b5cf21aa6af0398dcabe34022b1e","status":33,"value":{"message":"session not created: Missing or invalid capabilities\n  (Driver info: operadriver=99.0.4844.51 (d537ec02474b5afe23684e7963d538896c63ac77-refs/branch-heads/4844@{#875}),platform=Windows NT 10.0.22000 x86_64)"}}
_WD_CreateSession ==> Webdriver Exception (10) HTTP status = 200 : session not created: Missing or invalid capabilities
  (Driver info: operadriver=99.0.4844.51 (d537ec02474b5afe23684e7963d538896c63ac77-refs/branch-heads/4844@{#875}),platform=Windows NT 10.0.22000 x86_64)
```

Reply:
Take a look:
["alwaysMatch" can be omnited](https://github.com/Danp2/au3WebDriver/pull/285/commits/979fa7d53095ac1e513e7ad6c1fbc88bee7e76a8)

Reply:
You didn't exactly allow it to be omitted. You just shoved an empty alwaysMatch in there by default. ☹️

Reply:
it is fixed in later commit: 
[__WD_CapabilitiesNotation](https://github.com/Danp2/au3WebDriver/pull/285/commits/051662ea3b277995f0e17ea7892358f58d3b5a58)

Exactly here:
https://github.com/Danp2/au3WebDriver/pull/285#discussion_r846693318